lib/wordcounter.rb

```diff
@@ -0,0 +1,93 @@
+require 'delegate'
+
+class WordHistogram < SimpleDelegator
+  def initialize(*args)
+    super(Hash.new(0, *args))
+  end
+
+  def to_pretty_s
+    word_freq_pairs = self.sort_histogram
+
+    word_lengths = word_freq_pairs.map { |word, _| word }.map &:length
+    max_length = word_lengths.max
+
+    self.to_table_s(word_freq_pairs, middle_column_width: max_length)
+  end
+
+  protected
```

Comment:
Me initially misunderstanding how `self` modifies calls to private methods. i.e. `self.sort_histogram` vs. `sort_histogram`. The former fails when private but succeeds with protected.